src/piton/labelers/labeling_functions.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import collections
+import datetime
+import json
+import io
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import (
+    Any,
+    DefaultDict,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Set,
+    TextIO,
+    Tuple,
+    Union,
+)
+
+from .core import TimeHorizon, LabelingFunction, FixedTimeHorizonEventLF
+from .. import Event, Patient
+from ..extension import datasets as extension_datasets
+
+import numpy as np
+
+
+##########################################################
+# FixedTimeHorizonEventLF
+##########################################################
+
+class CodeLF(FixedTimeHorizonEventLF):
+    """
+        Applies a label based on a single code's occurrence over a fixed time horizon
+    """
+
+    def __init__(self, code: int, time_horizon: TimeHorizon):
+        """Label the code whose index in your Ontology is equal to `code`
+        """
+        self.code = code
+        self.time_horizon = time_horizon
+    
+    def get_time_horizon(self) -> TimeHorizon:
+        return self.time_horizon
+
+    def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
+        """Returns a list of datetimes corresponding to the start time of the Events
+            in this patient's timeline which have the exact same `code` as `self.code`
+        """
+        times: List[datetime.datetime] = []
+        for event in patient.events:
+            if event.code == self.code:
+                times.append(event.start)
+        return times
+
+class MortalityLF(CodeLF):
+    """
+        The mortality task is defined as predicting whether or not a
+        patient will die within the next `time_horizon` time.
+    """
+
+    def __init__(self, 
+                 ontology: extension_datasets.Ontology, 
+                 time_horizon: TimeHorizon):
+        CODE_DEATH_PREFIX = "Death Type/"
```

Comment:
This will need to be tested on the actual data/ actual OMOP schema. But we can do that later. Please add a TODO to this class to test on real data.